src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,374 @@
+                        children={placeholder}
+                        onClick={this.onPlaceholderClick}
+                    />
+                }
+                {!isTouch && !disabled && valueSet &&
+                    <Stepper
+                        className="stepper"
+                        onUp={this.onStepperUp}
+                        onDown={this.onStepperDown}
+                    />
+                }
+                {isTouch &&
+                    <label className="label">
+                        <input
+                            className="native-input"
+                            type="time"
+                            name={name}
+                            aria-label={label}
+                            value={this.getValue()}
+                            disabled={disabled}
+                            onFocus={this.onNantiveInputFocus}
+                            onBlur={this.onBlur}
+                            onChange={this.onNativeInputChange}
+                        />
+                    </label>
+                }
+            </div>
+        );
+    }
+
+    private getValue(): string {
+        const {hh = 0, mm = 0, ampm} = this.state;
+        const hhString = formatTimeChunk(to24(Number(hh), ampm));
+        const mmString = formatTimeChunk(mm);
+        return `${hhString}:${mmString}`;
+    }
+
+    private select(segment: Segment) {
+        const input = this.segments[segment]!;
+        this.setState({
+            currentSegment: segment
+        }, () => {
+            input.focus();
+            if (isTimeSegment(segment)) {
+                (input as HTMLInputElement).select();
+            }
+        });
+    }
+    private moveSelection(step: number): boolean {
+        const index = segments.indexOf(this.state.currentSegment!);
+        const nextSegment = segments[index + step];
+        if (this.segments[nextSegment]) {
+            this.select(nextSegment);
+            return true;
+        }
+        return false;
+    }
+
+    private updateSegmentValue(name: Segment, value: string | Ampm): void {
+        this.setState({
+            [name as any]: value
+        }, () => {
+            if (isTimeSegment(name)) {
+                this.select(name);
+            }
+            this.commit();
+        });
+    }
+
+    private toggleAmpm(shouldSelect: boolean): void {
+        this.updateSegmentValue('ampm', ampmSwitch[this.state.ampm]);
+        if (shouldSelect) {
+            this.segments.ampm!.focus();
+        } else {
+            this.segments.ampm!.blur();
+        }
+    }
+
+    private changeValue(step: number): void {
+        const {currentSegment, ampm} = this.state;
+        if (!isTimeSegment(currentSegment)) {
+            return this.toggleAmpm(true);
+        }
+        let newValue: any = Number(this.state[currentSegment] || 0) + step;
+        newValue = getCircularValue(currentSegment, newValue, ampm);
+        newValue = formatTimeChunk(newValue);
```

Comment:
use `formatTimeChunk ` inside `getCircularValue` so it will be 

```ts
const newValue = getCircularValue(
  currentSegment, 
  Number(this.state[currentSegment] || 0) + step, 
  ampm
);
this.updateSegmentValue(currentSegment, newValue);
```